product_category_company_favorite/models/product_category.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (C) 2023-Today: GRAP (<http://www.grap.coop/>)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class ProductCategory(models.Model):
+    _inherit = "product.category"
+
+    is_favorite = fields.Boolean(
+        string="Favorite",
+        company_dependent=True,
+        help="If this field is unchecked, the category will"
+        " be hidden when searching category in a drop-down list"
+        " like in the product form view."
+    )
+
+    @api.onchange("parent_id")
+    def onchange_parent_id_favorite(self):
+        if self.parent_id:
+            self.is_favorite = self.parent_id.is_favorite
+
+    @api.model
+    def create(self, vals):
+        category = super().create(vals)
+        # Configure is_favorite for all the other companies
+        company_ids = self.env["res.company"].with_context(
+            active_test=False
+        ).search([("id", "!=", self.env.user.company_id.id)]).ids
+        for company_id in company_ids:
+            ctx_category = category.with_context(force_company=company_id)
+            if ctx_category.parent_id:
+                # We inherit the contextual configuration of the parent category, if any
+                ctx_category.is_favorite = ctx_category.parent_id.is_favorite
```

Comment:
As stated by codecov, I think a quite important test is missing here: creation of a category that is a child of favorite=True category which should set the created child category favorite attribute to true.

Reply:
improved ``test_20_create_new_category_not_favorite`` and ``test_21_create_new_category_favorite``